### ISSUE TITLE
Float initialization with integer literal regression

### ISSUE THREAD

**Issue by mackrol** [taichi-dev/taichi#4451]
Regression w.r.t. to 0.8.11, in 0.9.0 float variable cannot be initialized with integer literal.
```py
import taichi as ti
ti.init(arch = ti.gpu)

@ti.kernel
def float_init():
    x : ti.f32 = 0 
    for _ in range(1):
        pass

float_init()
```
Results:
```
[Taichi] version 0.9.0, llvm 10.0.0, commit 0c432a37, win, python 3.9.5
[Taichi] Starting on arch=cuda
Traceback (most recent call last):

  File "D:\Devel\moita\moita\src\float_init.py", line 10, in <module>
    float_init()

  File "D:\Devel\WinPython-3.9\python-3.9.5.amd64\lib\site-packages\taichi\lang\kernel_impl.py", line 732, in wrapped
    raise type(e)('\n' + str(e)) from None

TaichiCompilationError: 
On line 6 of file "D:\Devel\moita\moita\src\float_init.py", in float_init:
    x : ti.f32 = 0
    ^^^^^^^^^^^^^^
Traceback (most recent call last):
  File "D:\Devel\WinPython-3.9\python-3.9.5.amd64\lib\site-packages\taichi\lang\ast\ast_transformer_utils.py", line 20, in __call__
    return method(ctx, node)
  File "D:\Devel\WinPython-3.9\python-3.9.5.amd64\lib\site-packages\taichi\lang\ast\ast_transformer.py", line 39, in build_AnnAssign
    node.ptr = ASTTransformer.build_assign_annotated(
  File "D:\Devel\WinPython-3.9\python-3.9.5.amd64\lib\site-packages\taichi\lang\ast\ast_transformer.py", line 66, in build_assign_annotated
    var = impl.make_constant_expr(value, anno)
  File "D:\Devel\WinPython-3.9\python-3.9.5.amd64\lib\site-packages\taichi\lang\util.py", line 216, in wrapped
    return func(*args, **kwargs)
  File "D:\Devel\WinPython-3.9\python-3.9.5.amd64\lib\site-packages\taichi\lang\impl.py", line 416, in make_constant_expr
    _check_in_range(to_numpy_type(constant_dtype), val)
  File "D:\Devel\WinPython-3.9\python-3.9.5.amd64\lib\site-packages\taichi\lang\impl.py", line 386, in _check_in_range
    iif = np.iinfo(npty)
  File "D:\Devel\WinPython-3.9\python-3.9.5.amd64\lib\site-packages\numpy\core\getlimits.py", line 523, in __init__
    raise ValueError("Invalid integer data type %r." % (self.kind,))
ValueError: Invalid integer data type 'f'.
```

`0` worked in 0.8.11, in 0.9.0 only `0.0` works. Although strictly correct, this change breaks lots of scripts.

**Comment by qiao-bo** [comment]
cc @strongoier @lin-hitonami

**Comment by strongoier** [comment]
Thanks for reporting this! This is indeed a bug and I will help fix this in v0.9.1.
